### ISSUE TITLE
Objective-C support

### ISSUE THREAD

**Issue by yairchu** [fpco/inline-c#142]
What would it take to support Objective-C too?

**Comment by bitonic** [comment]
I don't know -- I've never used Objective-C.

If its tooling is close to C/C++'s it would probably be pretty easy.

**Comment by yairchu** [comment]
Ok I still don't know exactly how the internals work but I got it working!

Just changing `TH.LangC` to `TH.LangObjc` in `initialiseModuleState` makes Objective-C work.

Then I can do

```Haskell
cglCtx <- [C.block| void* {
    NSOpenGLContext* object = (NSOpenGLContext*) $(void* nsglCtx);
    return [object CGLContextObj];
}|]
```

Instead of

```Haskell
cglCtx <- [C.block| void* {
    id object = (id) $(void* nsglCtx);
    SEL selector = sel_registerName("CGLContextObj");
    Method method = class_getInstanceMethod(object_getClass(object), selector);
    IMP imp = method_getImplementation(method);
    return ((CGLContextObj (*)(id, SEL))imp)(object, selector);
}|]
```

How would we like it do work? In a separate module/package like C++? If you have a general design for how this should work then I can make a PR for it.

**Comment by bitonic** [comment]
The right way to do this is in a separate package following the pattern of
inline-c-cpp.

**Comment by bitonic** [comment]
Fixed by #143 .
